File: LayoutTests/http/tests/media/resources/hls/sub-playlist-with-cookie.php
<?php
# See the HLS ITEF spec: <http://tools.ietf.org/id/draft-pantos-http-live-streaming-12.txt>
header("Status: 200 OK");
header("HTTP/1.1 200 OK");
header("Connection: close");
header('Cache-Control: no-store, no-cache="set-cookie"\n');
header("Access-Control-Allow-Origin: *");
header("Etag: " . '"' . filesize(__FILE__) . "-" . filemtime(__FILE__) . '"');
header("Set-Cookie: TEST=test.ts; path=/media/resources/");
header("Content-Type: application/x-mpegurl");
header("Content-Length: " . filesize(__FILE__));

$chunkDuration = 6.0272;
$chunkCount = 4;

function println($string) { return print($string . PHP_EOL); }

println("#EXTM3U");
println("#EXT-X-TARGETDURATION:7");
println("#EXT-X-VERSION:4");
println("#EXT-X-MEDIA-SEQUENCE:" . (time() / $chunkDuration % 100));

$time = time();
$time = $time - $time % $chunkDuration;

for ($i = 0; $i < $chunkCount; ++$i) {
    $time += 6;
    println("#EXT-X-PROGRAM-DATE-TIME:" . gmdate("c", $time));
    println("#EXTINF:" . $chunkDuration . ",");
    println("../video-cookie-check-cookie.php");
}
?>

File: LayoutTests/http/tests/media/resources/hls/sub-playlist-with-cookie.py
#!/usr/bin/env python3

# See the HLS ITEF spec: <http://tools.ietf.org/id/draft-pantos-http-live-streaming-12.txt>

import os
import sys
from datetime import datetime

sys.stdout.write(
    'status: 200\r\n'
    'Connection: close\r\n'
    'Cache-Control: no-store, no-cache="set-cookie"\\n\r\n'
    'Access-Control-Allow-Origin: *\r\n'
    'Pragma: no-cache\r\n'
    'Etag: "{size}-{mtime}"\r\n'
    'Set-Cookie: TEST=test.ts; path=/media/resources/\r\n'
    'Content-Type: application/x-mpegurl\r\n\r\n'.format(size=os.path.getsize(__file__), mtime=str(os.stat(__file__).st_mtime).split('.')[0])
)

chunk_duration = 6.0272
chunk_count = 4

sys.stdout.write(
    '#EXTM3U\n'
    '#EXT-X-TARGETDURATION:7\n'
    '#EXT-X-VERSION:4\n'
    '#EXT-X-MEDIA-SEQUENCE:{}\n'.format(int(datetime.utcnow().timestamp() / chunk_duration) % 100)
)

time = datetime.utcnow()
time = time.timestamp() - time.timestamp() % chunk_duration

for _ in range(0, chunk_count):
    sys.stdout.write(
        '#EXT-X-PROGRAM-DATE-TIME:{}T{}+00:00\n'
        '#EXTINF:{},\n'
        '../video-cookie-check-cookie.py\n'.format(datetime.fromtimestamp(time).strftime('%Y-%m-%d'), datetime.fromtimestamp(time).strftime('%H:%M:%S'), chunk_duration)
    )
